src/nvim/fileio.c

```diff
@@ -1737,9 +1737,17 @@ readfile (
   xfree(buffer);
 
   if (read_stdin) {
-    /* Use stderr for stdin, makes shell commands work. */
     close(0);
+#ifndef WIN32
+    // On Unix, use stderr for stdin, makes shell commands work.
     ignored = dup(2);
```

Comment:
In Vim this is guarded by `HAVE_DUP`, which is defined for Windows (`src/os_dos.h` in Vim's source).

https://github.com/vim/vim/blob/e7499ddc33508d3d341e96f84a0e7b95b2d6927c/src/fileio.c#L2397-L2398

I wonder why it doesn't work in Nvim.